File: PythonProject/Week 2/models/mark.py
class Mark:
    def __init__(self, record_id, student_id, subject_id, semester, marks_obtained, max_marks, exam_date):
        self.record_id = record_id
        self.student_id = student_id
        self.subject_id = subject_id
        self.semester = int(semester)
        self.marks_obtained = float(marks_obtained)
        self.max_marks = int(max_marks)
        self.exam_date = exam_date

    @property
    def percentage(self):
        return (self.marks_obtained / self.max_marks * 100) if self.max_marks > 0 else 0

    def __repr__(self):
        return f"Mark({self.record_id}, {self.student_id}, {self.subject_id}: {self.marks_obtained}/{self.max_marks})"

File: Java/Week_3-4/src/models/Mark.java
package models;

public class Mark
{
    private final String studentID;
    private final String subjectID;
    private final int semester;
    private final double marksObtained;
    private final double maxMarks;

    public Mark(String studentID, String subjectID, int semester, double marksObtained, double maxMarks)
    {
        this.studentID = studentID;
        this.subjectID = subjectID;
        this.semester = semester;
        this.marksObtained = marksObtained;
        this.maxMarks = maxMarks;
    }

    public String getStudentID(){return studentID;}
    public String getSubjectID(){return subjectID;}
    public int getSemester(){return semester;}
    public double getMarksObtained(){return marksObtained;}
    public double getMaxMarks(){return maxMarks;}

    double getPercentage()
    {return (marksObtained / maxMarks) * 100.0;}
}
